From e47d3fc313197ea9d44f6546bad430a7feed63d5 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:45 -0400
Subject: [tools] lib/traceevent: Unregister handler when scsi plugin is unloaded

Message-id: <1408953210-26343-278-git-send-email-jolsa@redhat.com>
Patchwork-id: 88936
O-Subject: [PATCH RHEL7.1 BZ1133083 277/322] tools lib traceevent: Unregister handler when scsi plugin is unloaded
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit a157112cfc67b3889f6493933cbd32620aa4be18
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Jan 16 11:31:17 2014 +0900

description
===========
The function handler should be unregistered when the plugin is unloaded
otherwise it'll try to access invalid memory.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/plugin_scsi.c b/tools/lib/traceevent/plugin_scsi.c
index 7ef16cc..eda326f 100644
--- a/tools/lib/traceevent/plugin_scsi.c
+++ b/tools/lib/traceevent/plugin_scsi.c
@@ -421,3 +421,9 @@ int PEVENT_PLUGIN_LOADER(struct pevent *pevent)
            PEVENT_FUNC_ARG_VOID);
  return 0;
 }
+
+void PEVENT_PLUGIN_UNLOADER(struct pevent *pevent)
+{
+ pevent_unregister_print_function(pevent, process_scsi_trace_parse_cdb,
+      "scsi_trace_parse_cdb");
+}
-- 
1.7.1